src/port/portindex.tcl

```diff
@@ -107,6 +129,24 @@ proc _open_port {portinfo_name portdir absportdir port_options_name {subport {}}
     set portinfo(portdir) $portdir
 }
 
+proc write_deleted_ports {} {
+    global qindex list_of_parsed_ports
+    foreach port [array names qindex] {
+        if {[lsearch -exact $list_of_parsed_ports $port] == -1 && [info exists qindex($port)] && [string trim $port] != ""} {
```

Comment:
Using a linear search here means a quadratic number of operations for the loop. That shouldn't be hard to avoid.

Reply:
Right.

Reply:
Okay, thank you.
Any views on using `lintersect3`?

> Call the first argument to intersect3 A and the second B. Then the first of the three results is A - B, the second result is A intersect B, and the third reuslt is B - A.

Reply:
No, you can simply use a map instead of maintaining a list of parsed ports where lookup time is O(1).

Reply:
Please see if I got your point correctly. Thanks for bearing with me.